common/logisticspipes/proxy/specialinventoryhandler/StorageDrawersInventoryHandler.java

```diff
@@ -200,11 +248,21 @@ public int getSizeInventory() {
 
 	@Override
 	public ItemStack getStackInSlot(int i) {
+		if (!_drawer.isDrawerEnabled(i))
+			return null;
+
 		return _drawer.getDrawer(i) != null ? _drawer.getDrawer(i).getStoredItemCopy() : null;
 	}
 
 	@Override
 	public ItemStack decrStackSize(int i, int j) {
-		return _drawer.takeItemsFromSlot(i, j);
+		if (!_drawer.isDrawerEnabled(i))
+			return null;
+
+		IDrawer drawer = _drawer.getDrawer(i);
+		if (drawer == null)
+			return null;
+
+		return getMultipleItems(ItemIdentifier.get(drawer.getStoredItemPrototype()), j);
```

Comment:
This can't be relaied to getMultipleItems();
It needs to access the slot i and not the complete inventory.